src/swap/SwapScreen.tsx

```diff
@@ -521,15 +566,15 @@ export function SwapScreen({ route }: Props) {
       newFromToken = selectedToken
       newSwitchedToNetworkId =
         toToken && toToken.networkId !== newFromToken.networkId ? newFromToken.networkId : null
-      if (newSwitchedToNetworkId) {
+      if (newSwitchedToNetworkId && !allowCrossChainSwaps) {
```

Comment:
should we include the check for allowCrossChainSwaps to derive `newSwitchedToNetworkId`? there is a `clearQuote` later in the function if the `newSwitchedToNetworkId` is defined, i think you don't want that?

Reply:
Ahh yeah, if we add `&& !allowCrossChainSwaps` to the condition for setting `newSwitchedToNetworkId`, we can catch that `clearQuote` case and also remove the `!allowCrossChainSwaps` conditions from the existing `if` statements.